packages/loot-core/src/shared/arithmetic.test.ts

```diff
@@ -41,4 +42,36 @@ describe('arithmetic', () => {
   test('respects current number format', () => {
     expect(evalArithmetic('1,222.45')).toEqual(1222.45);
   });
+
+  test('agnostic to decimal separator with dot-comma format', () => {
+    setNumberFormat({ format: 'dot-comma', hideFraction: false });
+    expect(evalArithmetic('1.222,4')).toEqual(1222.4);
+    expect(evalArithmetic('1.222.4')).toEqual(1222.4);
+    expect(evalArithmetic('1.222,45')).toEqual(1222.45);
+    expect(evalArithmetic('1.222.45')).toEqual(1222.45);
+  });
+
+  test('agnostic to decimal separator with comma-dot format', () => {
+    setNumberFormat({ format: 'comma-dot', hideFraction: false });
+    expect(evalArithmetic('1,222,45')).toEqual(1222.45);
+    expect(evalArithmetic('1,222.45')).toEqual(1222.45);
+    expect(evalArithmetic('1,222,4')).toEqual(1222.4);
+    expect(evalArithmetic('1,222.4')).toEqual(1222.4);
+  });
+
+  test('agnostic to decimal separator with space-comma format', () => {
+    setNumberFormat({ format: 'space-comma', hideFraction: false });
+    expect(evalArithmetic('1 222,45')).toEqual(1222.45);
+    expect(evalArithmetic('1 222.45')).toEqual(1222.45);
+    expect(evalArithmetic('1 222,4')).toEqual(1222.4);
+    expect(evalArithmetic('1 222.4')).toEqual(1222.4);
+  });
+
+  test('agnostic to decimal separator with space-dot format', () => {
+    setNumberFormat({ format: 'space-dot', hideFraction: false });
+    expect(evalArithmetic('1 222,45')).toEqual(1222.45);
+    expect(evalArithmetic('1 222.45')).toEqual(1222.45);
+    expect(evalArithmetic('1 222,4')).toEqual(1222.4);
+    expect(evalArithmetic('1 222.4')).toEqual(1222.4);
```

Comment:
Can you add these tests to the non-space format and the other tests to this format? It would be nice to have that behavior clarified.

Reply:
The way https://github.com/actualbudget/actual/blob/88f416b412d45ccf819d467fee69431de963d7ec/packages/loot-core/src/shared/arithmetic.ts#L50 works, space-comma additionally supports dots for thousands separator and space-dot supports commas. Is that correct, or did you want to make thousand separators agnostic too?

Reply:
Updated and added more tests.